### PR TITLE
Carousel - Adding accessibility layout

### PR DESCRIPTION
**Pull request by Inbal-Tish**
Use new prop 'allowAccessibilityLayout' to re-layout Carousel when the screen reader is on.

### CODE REVIEW

File: src/components/carousel/index.js [hunk 1]
```diff
@@ -75,7 +75,11 @@ export default class Carousel extends BaseComponent {
     /**
      * will block multiple pages scroll (will not work with 'pageWidth' prop)
      */
-    pagingEnabled: PropTypes.bool
+    pagingEnabled: PropTypes.bool,
+    /**
+     * Whether to layout Carousel for accessibility
+     */
+    allowAccessibleLayout: PropTypes.bool
```
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/components/carousel/index.js]: The reason is that Carousel is used in various ways and setting its unique accessible layout to be default, when VoiceOver, is on will make all Carousels vertical. For example, our demo app, where the main screen is made of a Carousel which you don't necessarily want to change the layout in this case.

File: src/components/carousel/index.js [hunk 2]
```diff
@@ -98,8 +102,13 @@ export default class Carousel extends BaseComponent {
       currentPage: this.shouldUsePageWidth() ? this.getCalcIndex(props.initialPage) : props.initialPage,
       currentStandingPage: props.initialPage,
       pageWidth: defaultPageWidth,
-      initialOffset: {x: presenter.calcOffset(props, {currentPage: props.initialPage, pageWidth: defaultPageWidth})}
+      initialOffset: {x: presenter.calcOffset(props, {currentPage: props.initialPage, pageWidth: defaultPageWidth})},
+      screenReaderEnabled: false
```
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/components/carousel/index.js]: It's not in the defaultProps, it's in this.state object...

File: src/components/carousel/index.js [hunk 3]
```diff
@@ -98,8 +102,13 @@ export default class Carousel extends BaseComponent {
...
     };
+
+    AccessibilityInfo.isScreenReaderEnabled().then((screenReaderEnabled) => {
```
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/components/carousel/index.js]: This was done before the addition of the const. I merged master to this branch but I can't test the accessibility constants feature so I don't want to make this change.

File: src/components/carousel/index.js [hunk 4]
```diff
@@ -190,6 +199,10 @@ export default class Carousel extends BaseComponent {
     this.setState(update);
   };
 
+  shouldAllowAccessibilityLayout() {
+    return this.props.allowAccessibleLayout && this.state.screenReaderEnabled;
```
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/components/carousel/index.js]: Same here. I can't make this change if I can't test it at the moment

File: src/components/carousel/index.js [hunk 5]
```diff
@@ -236,9 +249,16 @@ export default class Carousel extends BaseComponent {
...
     if (child) {
       const paddingLeft = this.shouldUsePageWidth() ? this.props.itemSpacings : undefined;
-  
+      const index = Number(key);
+      const length = presenter.getChildrenLength(this.props);
+
       return (
         <View style={{width: this.state.pageWidth, paddingLeft}} key={key} collapsable={false}>
+          {this.shouldAllowAccessibilityLayout() && !Number.isNaN(index) &&
+            <View style={{position: 'absolute', width: 1}}>
```
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/components/carousel/index.js]: Done

File: src/components/carousel/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/components/carousel/index.js]: By the way, if there shouldn't be a change in runtime there is no use for the listener there. You can just read once: 
`AccessibilityInfo.isScreenReaderEnabled().then((screenReaderEnabled) => {
      constants.accessibility.isScreenReaderEnabled = screenReaderEnabled;
 });`

File: src/components/carousel/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/components/carousel/index.js]: weird. it showed it as if it was in the defaultProps. 
Anyway, we don't need it now cause there's no need to save it in the state

File: src/components/carousel/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/components/carousel/index.js]: We kept the listener for QA reasons (it's not a lot of code so we kept it)
The reason for it so people can toggle voice over before they enter a screen (and so they won't have to restart the entire app)

regarding this
> This was done before the addition of the const. I merged master to this branch but I can't test the accessibility constants feature so I don't want to make this change.
 
I know, we just merged the Constants change yesterday.
Regarding testing, I'm fake testing by setting true hard-coded in the constants.

File: src/components/carousel/index.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (ethanshar) [reply src/components/carousel/index.js]: See what I wrote in the previous comment. set hard-coded true for the constants only when testing.

[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/components/carousel/index.js]: Done

File: src/components/carousel/index.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/components/carousel/index.js]: Done

File: src/components/carousel/index.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (Inbal-Tish) [reply src/components/carousel/index.js]: Done

**Pull request merged**